lighthouse-core/test/report/proto-test.js

```diff
@@ -11,6 +11,19 @@ const preprocessor = require('../../lib/proto-preprocessor.js');
 
 /* eslint-env jest */
 
+/* Proto conversion turns empty summaries into null. This is OK,
+   and is handled in the PSI roundtrip just fine, but messes up the easy
+   jest sub-object matcher. So, we put the empty object back in its place.
+*/
+function coerceNullSummaries(inputJson) {
+  for (const audit of Object.values(inputJson.audits)) {
+    if (audit.details && audit.details.summary === null) {
+      audit.details.summary = {};
+    }
+  }
+}
+coerceNullSummaries(roundTripJson);
```

Comment:
This null coercion has been moved outside of the `round trip JSON comparison to everything` describe block.

After regenerating the sample JSON via protobuf (`yarn update:sample-json`), a couple of additional summary fields are transformed into nulls:

https://github.com/GoogleChrome/lighthouse/pull/10072/files#diff-eaffc3a33b9584cf049b4d727638e15aR1193

This resulted in failures during the `round trip JSON comparison subsets` describe block, which checks each individual audit result for equality, but did not apply the same field coercion rules.  Both blocks still seem valuable since one tests the 'entire document' equality, including top-level keys, and the other can pinpoint specific audits which fail.

Reply:
This is actually due to having an older version of protobuf. If you update, you'll see that you can remove this code.

There was a version assertion, but it was removed in https://github.com/GoogleChrome/lighthouse/pull/8867 .

Reply:
@connorjclark Aha, interesting - I tried this with `3.6.1` initially - I *thought* I had upgraded to `3.11.1` after that, but it's very possible I didn't regenerate the samples correctly.  Just trying to unwind what I did there and re-apply.  Would be nice not to have to rearrange/modify this.

Reply:
@connorjclark Hrm; running `yarn update:sample-json` locally with `protoc==3.11.1` still results in the `summary` field values in the round-trip JSON (like [this one](https://github.com/GoogleChrome/lighthouse/pull/10072/files#diff-eaffc3a33b9584cf049b4d727638e15aR1193)) being turned into `null` values.  I'm going to try `3.7.1` to compare.

Reply:
@connorjclark Finally found the problem I was having locally:

Even though I was using `protoc>=3.7.1`, I had an old `python-protobuf` package installed on the system and it was being used during the [json_roundtrip_via_proto.py](https://github.com/GoogleChrome/lighthouse/blob/c4ab6a0759bbd84f10ad08ead195885c2b3ddf1d/proto/scripts/json_roundtrip_via_proto.py#L5-L6) step.

That took a bit of tracking down, but makes sense as the cause.  Just confirming that upgrading that clears things up completely..

Reply:
Yep, confirmed -- was able to run `yarn update:sample-json` as a no-op from 6f539628076a5c72144aebfeab5255a3687e5695 after correctly updating all `protobuf` dependencies.